File: day_2/rotate_array.py
def rotateArray(arr, k=0):
  arr_length = len(arr)
  if arr_length == 0:
    return []
  if k == 0:
    return arr
  if k % arr_length == 0:
    return arr
  
  k %= arr_length

  while(k > 0):
    val = arr[arr_length - 1]
    arr.pop()
    arr.insert(0, val)
    k -= 1

  return arr


a = [0, 1, 2, 3, 4, 5, 7]
b = []
c = [6, 8, 3]


print(rotateArray(a, 2))
print(rotateArray(b, 2))
print(rotateArray(c))

File: day_2/rotateArray.js
function rotate(arr, k) {
  let arrLength = arr.length;
  if (arrLength === null) return [];

  if (k === 0) return arr;
  if (k % arrLength === 0) return arr;

  k %= arrLength;

  while(k > 0) {
    let value = arr[arrLength - 1];
    arr.pop();
    arr.unshift(value);
    k--;
  }

  return arr
}

const a = [0, 1, 2, 3, 4, 5, 7];
const b = [];
const c = [6, 8, 3];


console.log(rotate(a, 2));
console.log(rotate(b, 2));
console.log(rotate(c));